DOCS 5517: updating k8s configuration page (#18657)
DataDog/documentation: content/en/containers/kubernetes/configuration.md

After you have installed the Datadog Agent in your Kubernetes environment, you may choose additional configuration options.

### Enable Datadog to collect:
- [Traces (APM)](#enable-apm-and-tracing)
- [Kubernetes events](#enable-kubernetes-event-collection)

> - [NPM](#enable-npm)

Reviewer: ```suggestion
- [NPM](#enable-npm-collection)
```

- [Logs](#enable-log-collection)
- [Processes](#enable-process-collection)

### Other capabilities
- [Datadog Cluster Agent](#datadog-cluster-agent)
- [Integrations](#integrations)
- [Containers view](#containers-view)
- [Orchestrator Explorer](#orchestrator-explorer)

> - [External metrics server](#external-metrics-server)

Reviewer: ```suggestion
- [External metrics server](#custom-metrics-server)
```
(the sub-title link goes to "custom metrics server" on the page, didn't know if you also want to change the name here from "External" to "Custom" as well ?

{{% tab "Helm" %}}
Update your `datadog-values.yaml` with the following configuration:

```yaml
datadog:
  # (...)
  logs:
    enabled: true
    containerCollectAll: true
```

Then upgrade your Helm chart:

```shell
helm upgrade -f datadog-values.yaml <RELEASE_NAME> datadog/datadog
```
{{% /tab %}}
{{< /tabs >}}

For more information, see [Kubernetes log collection][17].

## Enable process collection

{{< tabs >}}
{{% tab "Datadog Operator" %}}
In your `datadog-agent.yaml`, set `features.liveProcessCollection.enabled` to `true`.

```yaml
apiVersion: datadoghq.com/v2alpha1
kind: DatadogAgent
metadata:
  name: datadog
spec:
  global:
    credentials:
      apiKey: <DATADOG_API_KEY>

  features:
    liveProcessCollection:
      enabled: true
```

Then apply the new configuration:

```shell
kubectl apply -n $DD_NAMESPACE -f datadog-agent.yaml
```

{{% /tab %}}
{{% tab "Helm" %}}
Update your `datadog-values.yaml` with the following configuration:

```yaml
datadog:
  # (...)
  processAgent:
    enabled: true
    processCollection: true
```

Then upgrade your Helm chart:

```shell
helm upgrade -f datadog-values.yaml <RELEASE_NAME> datadog/datadog
```
{{% /tab %}}
{{< /tabs >}}

> 

Reviewer: Did you also want to link to https://docs.datadoghq.com/infrastructure/process/ page to keep the same format/flow as the other sections that link to their relative functionality pages ? 
i.e. "For more information, see [Live Processes](https://docs.datadoghq.com/infrastructure/process/).

  clusterAgentTokenSecret: 
    secretName: <SECRET_NAME>
    keyName: <KEY_NAME>
```

**Note**: When set manually, this token must be 32 alphanumeric characters.

Then apply the new configuration:

```shell
kubectl apply -n $DD_NAMESPACE -f datadog-agent.yaml
```

{{% /tab %}}
{{% tab "Helm" %}}

Helm chart v2.7.0+ enables the Cluster Agent by default.

For verification, ensure that `clusterAgent.enabled` is set to `true` in your `datadog-values.yaml`:

```yaml
clusterAgent:
  enabled: true
```

Helm automatically generates a random token in a Kubernetes `Secret` to be shared by the Datadog Agent and Cluster Agent for secure communication. 

You can manually specify this token in the `clusterAgent.token` field in your `datadog-agent.yaml`:

```yaml
clusterAgent:
  enabled: true
  token: <DATADOG_CLUSTER_AGENT_TOKEN>
```

Alternatively, you can specify this token by referencing the name of an existing `Secret`, where the token is in a key named `token`:

```yaml
clusterAgent:
  enabled: true
  tokenExistingSecret: <SECRET_NAME>
```

{{% /tab %}}
{{< /tabs >}}

For more information, see the [Datadog Cluster Agent documentation][2].

## Custom metrics server

> To use the Cluster Agent's [custom metrics server][22] feature, you must supply a Datadog app key and enable the metrics provider.

Reviewer: ```suggestion
To use the Cluster Agent's [custom metrics server][22] feature, you must supply a Datadog [application key](account_management/api-app-keys/#application-keys) and enable the metrics provider.
```

{{< tabs >}}
{{% tab "Datadog Operator" %}}

> In `datadog-agent.yaml`, supply an app key and set `features.externalMetricsServer.enabled` to `true`.

Reviewer: ```suggestion
In your `datadog-agent.yaml`, supply an `appKey` and set `features.externalMetricsServer.enabled` to `true`.
```

```yaml
apiVersion: datadoghq.com/v2alpha1
kind: DatadogAgent
metadata:
  name: datadog
spec:
  global:
    credentials:
      apiKey: <DATADOG_API_KEY>
      appKey: <DATADOG_API_KEY>

  features:
    externalMetricsServer:
      enabled: true
```

Then apply the new configuration:

```shell
kubectl apply -n $DD_NAMESPACE -f datadog-agent.yaml
```
{{% /tab %}}
{{% tab "Helm" %}}

> In `datadog-values.yaml`, supply an app key and set `clusterAgent.metricsProvider.enabled` to `true`.

Reviewer: ```suggestion
In `datadog-values.yaml`, supply an `appKey` and set `clusterAgent.metricsProvider.enabled` to `true`.
```

| `datadog.logLevel` | Sets logging verbosity. This can be overridden by the container. Valid log levels are: `trace`, `debug`, `info`, `warn`, `error`, `critical`, and `off`. Default: `info`. |
| `registry` | Image registry to use for all Agent images. Default: `gcr.io/datadoghq`. |
| `datadog.site` | Sets the Datadog [intake site][1] to which Agent data is sent. Your site is {{< region-param key="dd_site" code="true" >}}. (Ensure the correct SITE is selected on the right). |
| `datadog.tags` | A list of tags to attach to every metric, event, and service check collected. |

For a complete list of environment variables for the Helm chart, see the [full list of options][2] for `datadog-values.yaml`.

[1]: /getting_started/site
[2]: https://github.com/DataDog/helm-charts/tree/main/charts/datadog#all-configuration-options
{{% /tab %}}
{{< /tabs >}}

## Configure DogStatsD

DogStatsD can send custom metrics over UDP with the StatsD protocol. **DogStatsD is enabled by default by the Datadog Operator and Helm**. See the [DogStatsD documentation][19] for more information.

> You can use the following environment variables to configure DogStatsD.

Reviewer: Is there supposed to be an Environment variable section here for DogsStatsD ?

Exclude containers from logs collection, metrics collection, and Autodiscovery. Datadog excludes Kubernetes and OpenShift `pause` containers by default. These allowlists and blocklists apply to Autodiscovery only; traces and DogStatsD are not affected. These environment variables support regular expressions in their values.

> | Helm Parameter |  Description |

Reviewer: It doesn't appear Operator has specific functional envars for include/exclude containers ?
